examples/cpustat.lua

```diff
@@ -0,0 +1,45 @@
+--
+-- SPDX-FileCopyrightText: (c) 2025 Enderson Maia <endersonmaia@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+-- Simple example demonstrating linux.cpustat() usage
+
+local linux = require("linux")
+
+-- Get number of CPUs
+local num_cpus = linux.nproc()
+print(string.format("System has %d online CPUs\n", num_cpus))
+
+local fmt = [[
+CPU %d statistics:
+  user:       %d
+  nice:       %d
+  system:     %d
+  idle:       %d
+  iowait:     %d
+  irq:        %d
+  softirq:    %d
+  steal:      %d
+  guest:      %d
+  guest_nice: %d
+
+]]
+
+-- Get stats for each CPU
+for i = 0, num_cpus - 1 do
+	local stats = linux.cpustat(i)
+	
+	print(string.format(fmt,
+		i,
+		stats.user,
+		stats.nice,
+		stats.system,
+		stats.idle,
+		stats.iowait,
+		stats.irq,
+		stats.softirq,
+		stats.steal,
+		stats.guest,
+		stats.guest_nice
+	))
```

Comment:
I think we don't need to break this in too many lines as this, right?
```suggestion
	print(string.format(fmt, i, stats.user, stats.nice, stats.system, stats.idle,
		stats.iowait, stats.irq, stats.softirq, stats.steal, stats.guest, stats.guest_nice))
```